backend/src/hatchling/metadata/core.py

```diff
@@ -462,7 +462,8 @@ def description(self) -> str:
             if not isinstance(description, str):
                 message = 'Field `project.description` must be a string'
                 raise TypeError(message)
-
+            if '\n' or '\r\n' in description:
+                description = description.replace('\r\n', ' ').replace('\n', ' ')
             self._description = description
```

Comment:
```suggestion
            self._description = ' '.join(description.splitlines())
```

Reply:
Wow, great improvement! My previous code was too terrible.

Reply:
No yours was good! I actually just did a similar thing in a different project so I just copied that code